raiden-ts/tests/unit/epics/monitor.spec.ts

```diff
@@ -241,119 +146,186 @@ describe('monitorRequestEpic', () => {
 
-    const promise = monitorRequestEpic(action$, state$, depsMock).toPromise();
-    action$.next(
-      udcDeposit.success(undefined, { totalDeposit: monitoringReward.mul(2) as UInt<32> }),
-    );
-
-    await receiveTransfer(10);
-    setTimeout(() => action$.complete(), 50);
+    await ensureChannelIsDeposited([raiden, partner], deposit);
+    await ensureChannelIsDeposited([partner, raiden], deposit);
+    await ensureTransferUnlocked([partner, raiden], amount);
+    await waitBlock();
 
-    await expect(promise).resolves.toBeUndefined();
+    expect(raiden.output).not.toContainEqual(
+      messageGlobalSend(
+        { message: expect.objectContaining({ type: MessageType.MONITOR_REQUEST }) },
+        expect.anything(),
+      ),
+    );
   });
 
   test('ignore: signing rejected not fatal', async () => {
-    expect.assertions(2);
+    expect.assertions(3);
 
-    const promise = monitorRequestEpic(action$, state$, depsMock).toPromise();
-    action$.next(
-      udcDeposit.success(undefined, { totalDeposit: monitoringReward.mul(2) as UInt<32> }),
+    const [raiden, partner] = await makeRaidens(2);
+    const monitoringReward = bigNumberify(5) as UInt<32>;
+    const error = new RaidenError(ErrorCodes.RDN_GENERAL_ERROR);
+    raiden.store.dispatch(
+      raidenConfigUpdate({
+        httpTimeout: 30,
+        monitoringReward,
+        caps: {
+          [Capabilities.NO_DELIVERY]: true,
+          [Capabilities.NO_MEDIATE]: true,
+          // 'noReceive' should be auto-disabled by 'getCaps$'
+        },
+        // rate=WeiPerEther == 1:1 to SVT
+        rateToSvt: { [token]: WeiPerEther as UInt<32> },
+      }),
     );
-
+    await ensureChannelIsDeposited([raiden, partner], deposit);
+    await ensureChannelIsDeposited([partner, raiden], deposit);
+    await ensureTransferPending([partner, raiden], amount);
     // fails only after transfer's signatures
-    const originalSign = depsMock.signer.signMessage;
+    const originalSign = raiden.deps.signer.signMessage;
     const signerSpy = jest
-      .spyOn(depsMock.signer, 'signMessage')
+      .spyOn(raiden.deps.signer, 'signMessage')
       .mockImplementation(async (message) => {
         if (signerSpy.mock.calls.length > 1) throw new Error('Signature rejected');
-        return originalSign.call(depsMock.signer, message);
+        return originalSign.call(raiden.deps.signer, message);
       });
 
-    await receiveTransfer(10);
-    setTimeout(() => action$.complete(), 100);
-
-    await expect(promise).resolves.toBeUndefined();
+    await ensureTransferUnlocked([partner, raiden], amount);
+    await waitBlock();
 
-    expect(signerSpy).toHaveBeenCalledTimes(2);
-    signerSpy.mockRestore();
+    expect(raiden.output).not.toContainEqual(
+      messageGlobalSend(
+        { message: expect.objectContaining({ type: MessageType.MONITOR_REQUEST }) },
+        expect.anything(),
+      ),
+    );
+    expect(last(raiden.output)).not.toEqual(raidenShutdown({ reason: error }));
```

Comment:
You want to be broad. Don't check only `last` (although it'd be safe to expect the shutdown would be the last action), but be broader that it output does `.not.toContainEqual(raidenShutdown(expect.anything()))`